vectordraw.py

```diff
@@ -67,9 +120,25 @@ def check_angle(check, vectors):
     y = vec.tip.y - vec.tail.y
     dot_product = x * math.cos(expected) + y * math.sin(expected)
     angle = math.degrees(math.acos(dot_product / vec.length))
-    if abs(angle) > tolerance:
+    return abs(angle) > tolerance
+
+def check_angle(check, vectors):
+    vec = vectors[check['vector']]
+    tolerance = check.get('tolerance', 2.0)
+    expected = math.radians(check['expected'])
+    if _check_angle(vec, expected, tolerance):
         return 'The angle of {} is incorrect. Your angle: {:.1f}'.format(vec.name, vec.angle)
 
+def check_segment_angle(check, vectors):
+    # Segments are not directed, so we must check the angle between the segment and
+    # the vector that represents it, as well as its opposite vector.
+    vec = vectors[check['vector']]
+    tolerance = check.get('tolerance', 2.0)
+    expected = math.radians(check['expected'])
+    if _check_angle(vec, expected, tolerance):
+        if _check_angle(vec.opposite(), expected, tolerance):
```

Comment:
@mtyaka nit: I would still use `and`. If you're concerned of `\`, maybe formatting it like this would look better?

    if (
        _check_angle(vec, expected, tolerance) and
        _check_angle(vec.opposite(), expected, tolerance)
    ):